{Release} Upgrade to Azure CLI 2.49.0 (#26479)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.49.0
++++++

**ACR**

* `az acr create`: Remove `Classic` from `sku` option (#26116)

**AKS**

> * [BREAKING CHANGE] `az aks create`: Specifying `--pod-cidr` with Azure CNI will return an error instead of logging a warning when not use `overlay` mode (#26237)

Reviewer: ```suggestion
* [BREAKING CHANGE] `az aks create`: Specify `--pod-cidr` with Azure CNI will return an error instead of logging a warning when not use `overlay` mode (#26237)
```

> * [BREAKING CHANGE] `az aks create`: Specifying `--pod-cidr` with Azure CNI will return an error instead of logging a warning when not use `overlay` mode (#26237)

Reviewer: ```suggestion
* [BREAKING CHANGE] `az aks create`: Specify `--pod-cidr` with Azure CNI will return an error instead of logging a warning when not use `overlay` mode (#26237)
```

> * [BREAKING CHANGE] `aks create`: Change the default value of `--enable-msi-auth-for-monitoring` to true and add check for airgap clouds (#26356)

Reviewer 2: ```suggestion
* [BREAKING CHANGE] `az aks create`: Change the default value of `--enable-msi-auth-for-monitoring` to true and add check for airgap clouds (#26356)
```

> * [BREAKING CHANGE] `aks create`: Change the default value of `--enable-msi-auth-for-monitoring` to true and add check for airgap clouds (#26356)

Reviewer 3: ```suggestion
* [BREAKING CHANGE] `az aks create`: Change the default value of `--enable-msi-auth-for-monitoring` to true and add check for airgap clouds (#26356)
```

> * `az aks update`: Support update user assigned control plane identity for parameter ` --assign-identity` (#25973)

Reviewer 3: ```suggestion
* `az aks update`: Support updating user assigned control plane identity for parameter `--assign-identity` (#25973)
```

* `az aks install-cli`: Add validation for installation path and update help message for parameters `--install-location` and `--kubelogin-install-location` (#26357)
* Fix #26353: `az aks install-cli`: Fix incorrect architecture detection on Darwin/arm64 (#26366)
* `az aks create/update`: Add parameter `--enable-azure-monitor-metrics` to enable managed prometheus (Azure Monitor Metrics Addon) (#26201)
* `az aks create/update`: Add parameter `--azure-monitor-workspace-resource-id` to store metrics for the managed prometheus addon (#26201)
* `az aks create/update`: Add parameter `--grafana-resource-id` to link the Azure Monitor Workspace with a Grafana instance for viewing metrics and dashboards (#26201)
* `az aks create/update`: Add parameter `--enable-windows-recording-rules` to enable windows recording rule groups on the Azure Monitor Workspace (by default they get created but are disabled) (#26201)

> * `az aks create/update`: Add parameter `--ksm-metric-labels-allow-list` to support the additional Kubernetes label keys that will be used in the resource' labels metric (#26201)
> * `az aks create/update`: Add parameter `--ksm-metric-annotations-allow-list` to support the Kubernetes annotations keys that will be used in the resource' labels metric (#26201)

Reviewer 4: ```suggestion
* `az aks create/update`: Add parameter `--ksm-metric-labels-allow-list` to support the additional Kubernetes label keys that will be used in the resource's labels metric (#26201)
* `az aks create/update`: Add parameter `--ksm-metric-annotations-allow-list` to support the Kubernetes annotations keys that will be used in the resource's labels metric (#26201)
```

* [BREAKING CHANGE] `az appconfig kv import`: Update feature name validation. Invalid feature flags will be skipped during import (#26079)
* [BREAKING CHANGE] `az appconfig`: Update default connection string resolution logic (#25120)

**App Service**

* `az functionapp create`: Add new parameter `--min-replicas` and `--max-replicas` to support minimum and maximum replicas (#26169)
* `az functionapp create`: Add new parameter `--registry-server` to support Centauri function app (#26307)
* `az functionapp create`: Update the default image to `mcr.microsoft.com` for Centauri (#26351)
* Fix #26445: `az webapp deploy`: Fix deployment failing with HTTP 400 (#26452)

**ARM**

* Fix #26216: `az bicep format`: Fix the TypeError `expected str, bytes or os.PathLike object, not bool` (#26249)
* Fix #26256: `az bicep publish/restore/generate-params`: Fix version checks without bicep installed (#26374)
* `az bicep publish`: Add new parameter `--force` to allow overwriting existing module (#26360)
* Fix #26352: `az ts create`: Fix for the TypeError `string indices must be integers` (#26363)

**Backup**

> * Add support for HANA HSR workload (#26368)

Reviewer: ```suggestion
* `az backup`: Add support for HANA HSR workload (#26368)
```

* `az container create`: Add new parameters for container security context for confidential ContainerGroupSku (#26273)

**Cosmos DB**

* [BREAKING CHANGE] `az cosmosdb create/update`: Rename `--enable-public-network true/false` to `--public-network-access ENABLED/DISABLED/SECUREDBYPERIMETER` (#26226)
* `az cosmosdb create/update`: Add `--continuous-tier` to support continuous backup tier (#26226)
* `az cosmosdb create/update`: Enable Partition Merge feature for CosmosDB (#26392)

**Eventhub**

* [BREAKING CHANGE] `az eventhubs namespace network-rule`: This command group is removed and replaced by `az eventhubs namespace network-rule-set` (#25792)
* [BREAKING CHANGE] `az eventhubs namespace network-rule add`: This command is removed and replaced by `az eventhubs namespace network-rule-set ip-rule/virtual-network-rule add` (#25792)
* [BREAKING CHANGE] `az eventhubs namespace network-rule remove`: This command is removed and replaced by `az eventhubs namespace network-rule-set ip-rule/virtual-network-rule remove` (#25792)
* [BREAKING CHANGE] `az eventhubs eventhub create/update`: Remove `--message-retention` parameter, it is replaced by `--retention-time-in-hours` (#26168)

> * [BREAKING CHANGE] `az eventhubs namespace application-group policy remove`: Renamed `--throttling-policy-config` to `--policy` and removed `metric-id` and `rate-limit-threshold` properties in it (#26032)

Reviewer 4: ```suggestion
* [BREAKING CHANGE] `az eventhubs namespace application-group policy remove`: Rename `--throttling-policy-config` to `--policy` and remove `metric-id` and `rate-limit-threshold` properties in it (#26032)
```

* `az eventhubs eventhub create/update`: Add `--cleanup-policy`, `--retention-time-in-hours` and `--tombstone-retention-time-in-hours` to support `Retention-Description` feature (#26168)

**IoT**

> *  `az iot hub create/update/delete`: Fix poller issues (#26296)

Reviewer 3: ```suggestion
* `az iot hub create/update/delete`: Fix poller issues (#26296)
```

**Key Vault**

* [BREAKING CHANGE] `az keyvault create`: `--retention-days` becomes required for MHSM creation (#26423)
* [BREAKING CHANGE] `az keyvault backup start`: The output will only contain `folderUrl` (#26422)
* [BREAKING CHANGE] `az keyvault restore start`: Nothing will return for successful run (#26422)
* [BREAKING CHANGE] `az keyvault role assignment delete`: Nothing will return for successful run (#26422)
* [BREAKING CHANGE] `az keyvault certificate show/set-attributes/import`: No longer return `x509CertificateProperties.basicConstraints`, `pending` (#26242)
* [BREAKING CHANGE] `az keyvault certificate contact delete`: Return an empty list instead of the deleted contact for consistency if the operation would remove the last contact (#26242)
* [BREAKING CHANGE] `az keyvault certificate issuer create`: `organizationDetails.zip` is no longer returned by serivce, use 0 as the default (#26242)
* `az keyvault security-domain upload`: Fix sd warpping keys with passwords (#26288)
* `az keyvault setting`: New command group to manage MHSM settings (#26422)

**Monitor**

* `az monitor`: Add new subgroup `account` to support managing monitor workspace (#26358)

> * `az monitor log-analytics workspace table create/update`: `--total-retention-time` fix upper validation (#26384)

Reviewer 3: ```suggestion
* `az monitor log-analytics workspace table create/update`: Max `--total-retention-time` changed from 2555 to 2556 (#26384)
```

> * `az monitor log-analytics workspace table create/update`: `--total-retention-time` fix upper validation (#26384)

Reviewer 5: * `az monitor log-analytics workspace table create/update`: `--total-retention-time` upper validation update from 2555 to 2556(#26384)

* [BREAKING CHANGE] `az network nic ip-config address-pool remove`: Deprecate the output of command (#26458)
* [BREAKING CHANGE] `az network nic ip-config inbound-nat-rule remove`: Deprecate the output of command (#26458)
* [BREAKING CHANGE] `az network lb address-pool tunnel-interface remove`: Deprecate the output of command (#26458)
* [BREAKING CHANGE] `az network cross-region-lb address-pool address remove`: Deprecate the output of command (#26458)
* `az network private-endpoint-connection`: Add provider `Microsoft.HardwareSecurityModules/cloudHsmClusters` (#26235)
* Fix #26248: `az network dns record-set cname set-record`: Declare TTL as integer type (#26260)
* Fix #26326: `az network vnet subnet update`: `--nat-gateway` cannot be set to null (#26365)
* Fix #26318: `az network vnet subnet create`: `--nsg` and `--route-table` cannot be used as name from Azure Stack (#26382)

**RDBMS**

* [BREAKING CHANGE] `az mysql/postgres flexible-server create/update`: Deprecate `Enabled` for `--high-availability` argument (#26276)
* `az mysql flexible-server restore/georestore/replica create`: Add public-access parameter for restore, replication and georestore (#26424)

**Service Bus**

> * [BREAKING CHANGE] `az servicebus georecovery-alias fail-over`: Removed `--parameters` argument (#26054)

Reviewer: ```suggestion
* [BREAKING CHANGE] `az servicebus georecovery-alias fail-over`: Remove `--parameters` argument (#26054)
```

> * [BREAKING CHANGE] `az servicebus georecovery-alias fail-over`: Removed `--parameters` argument (#26054)

Reviewer 4: ```suggestion
* [BREAKING CHANGE] `az servicebus georecovery-alias fail-over`: Remove `--parameters` argument (#26054)
```

* [BREAKING CHANGE] `az servicebus namespace network-rule`: This command group is removed and replaced by `az servicebus namespace network-rule-set` (#25719)
* [BREAKING CHANGE] `az servicebus namespace network-rule add`: This command is removed and replaced by `az servicebus namespace network-rule-set ip-rule/virtual-network-rule add` (#25719)
* [BREAKING CHANGE] `az servicebus namespace network-rule remove`: This is removed and replaced by by `az servicebus namespace network-rule-set ip-rule/virtual-network-rule remove` (#25719)
* [BREAKING CHANGE] `az servicebus queue update`: Remove deprecated parameters `--enable-partitioning`, `--enable-session` and `--duplicate-detection` (#26478)

**Service Connector**

* `az spring connection create`: Enable user-assigned managed identity for spring apps and deprecate Postgresql single server (#26259)

**SQL**

* [BREAKING CHANGE] `az sql mi link create`: Remove `--replication-mode` argument (#26400)
* `az sql elastic-pool`: Add `--preferred-enclave-type` argument (#25968)
* `az sql mi link update`: Fix update command to use PATCH api (#26400)

**Storage**

> * `az storage account blob-service-properties cors-rule`: New command group to manage blob cors rules (#26447)

Reviewer 4: ```suggestion
* `az storage account blob-service-properties cors-rule`: Add new command group to manage blob cors rules (#26447)
```
